src/main/java/tennisclub/entity/User.java

```diff
@@ -92,8 +91,7 @@ public boolean equals(Object o) {
         if (!(o instanceof User)) return false;
 
         User user = (User) o;
-
-        return getUsername().equals(user.getUsername());
+        return username != null && username.equals(user.username);
```

Comment:
I believe `getUsername` should be used here 👀

Reply:
@Fuyunori

The main point of this change is testing whether `username != null`. Before, calling `equals()` on a `Court` with `username == null` caused a `NullPointerException` to be thrown.

Another way to deal with it would be `Objects.equals(username, user.username)`. However, that returns `true` for 2 `null`s, which I do not think is correct in this case, i.e. 2 users without a username are **not** necessarily the same user.

The reason for accessing the attribute directly -- not via its getter -- is, well, mostly brevity and instinctive avoidance of unnecessary function calls. Is it an antipattern of some sort?

Reply:
I think in the lecture they stressed that in this case the `equals` and `hashCode` methods should use getters since hibernate might use a subclass of the entity as a proxy. Then the getters, overridden in the subclass, might contain some internal functionality that would not be called.

Reply:
@bigno78 Thank you for the remark; I will fix it then.

Reply:
Now that I think about it, `username` cannot be null as it's declared `@NotNull`. So maybe it is enough to only compare them using `equals`?

Reply:
@bigno78 The validation is done when persisting the entity. Therefore before persisting, it **can** be `null` (and still called `equals()` on).

Maybe this is paranoid-level defensive thinking. But at this point, I honestly do not see past the DAO layer (implementation-wise) and have little idea about the subsequent usage of these classes. And, well, better safe than sorry, huh? :slightly_smiling_face: